metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py

```diff
@@ -127,6 +138,41 @@ def get_subtype(self) -> DatasetSubTypes:
         return DatasetSubTypes.VIEW
 
 
+@dataclass
+class SnowflakeSemanticView(BaseView):
+    """Represents a Snowflake Semantic View with dimensions, facts, and metrics."""
+
+    columns: List[SnowflakeColumn] = field(default_factory=list)
+    tags: Optional[List[SnowflakeTag]] = None
+    column_tags: Dict[str, List[SnowflakeTag]] = field(default_factory=dict)
+    # Base tables that this semantic view is built on (for lineage)
+    base_tables: List[Tuple[str, str, str]] = field(
+        default_factory=list
+    )  # List of (db, schema, table) tuples
```

Comment:
We should have a class `SnowflakeTableIdentifier` akin to the BigQuery one, just a note.